azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -34,107 +34,145 @@ class AgentUpdateHandlerUpdateState(object):
 
-    def __check_if_agent_update_allowed_and_update_next_upgrade_times(self, requested_version):
-        """
-        requested version update:
-            update is allowed once per (as specified in the conf.get_autoupdate_frequency())
-            return false when we don't allow updates.
-        largest version update(self-update):
-            update is allowed once per (as specified in the conf.get_hotfix_upgrade_frequency() or conf.get_normal_upgrade_frequency())
-            return false when we don't allow updates.
-        """
-        now = datetime.datetime.now()
-
-        if self._is_requested_version_update:
-            if self.update_state.last_attempted_requested_version_update_time != datetime.datetime.min:
-                next_attempt_time = self.update_state.last_attempted_requested_version_update_time + datetime.timedelta(seconds=conf.get_autoupdate_frequency())
-            else:
-                next_attempt_time = now
-
-            if next_attempt_time > now:
-                return False
-            # The time limit elapsed for us to allow updates.
-            self.update_state.last_attempted_requested_version_update_time = now
-            return True
-        else:
-            next_hotfix_time, next_normal_time = self.__get_next_upgrade_times(now)
-            upgrade_type = self.__get_agent_upgrade_type(requested_version)
-
-            if (upgrade_type == AgentUpgradeType.Hotfix and next_hotfix_time <= now) or (
-                    upgrade_type == AgentUpgradeType.Normal and next_normal_time <= now):
-                # Update the last upgrade check time even if no new agent is available for upgrade
-                self.update_state.last_attempted_hotfix_update_time = now
-                self.update_state.last_attempted_normal_update_time = now
-                return True
-            return False
-
-    def __should_agent_attempt_manifest_download(self):
-        """
-        The agent should attempt to download the manifest if
-        the agent has not attempted to download the manifest in the last 1 hour
-        """
-        now = datetime.datetime.now()
-
-        if self.update_state.last_attempted_manifest_download_time != datetime.datetime.min:
-            next_attempt_time = self.update_state.last_attempted_manifest_download_time + datetime.timedelta(seconds=conf.get_autoupdate_frequency())
+        # restore the state of rsm update
+        if not os.path.exists(self._get_rsm_version_state_file()):
+            self._is_version_from_rsm = False
+            self._is_vm_enabled_for_rsm_upgrades = False
         else:
-            next_attempt_time = now
-
-        if next_attempt_time > now:
-            return False
-        self.update_state.last_attempted_manifest_download_time = now
-        return True
+            self._is_version_from_rsm = self._get_is_version_from_rsm()
+            self._is_vm_enabled_for_rsm_upgrades = self._get_is_vm_enabled_for_rsm_upgrades()
```

Comment:
get_autoupdate_frequency() and get_autoupdate_frequency() already return False if **not os.path.exists(self._get_rsm_version_state_file()):**

I don't think you need the if/else condition here